checks.d/redisdb.py

```diff
@@ -233,9 +243,64 @@ def _check_db(self, instance, custom_tags=None):
                             slave_tags.append('slave_{0}:{1}'.format(slave_tag, info[key][slave_tag]))
                     slave_tags.append('slave_id:%s' % key.lstrip('slave'))
                     self.gauge('redis.replication.delay', delay, tags=slave_tags)
+    
+
+    def _check_slowlog(self, instance, custom_tags):
+        """Retrieve length and entries from Redis' SLOWLOG
+
+        This will parse through all entries of the SLOWLOG and select ones
+        within the time range between the last seen entries and now
+
+        """
+
+        max_slow_entries = "slowlog-max-len"
+        conn = self._get_conn(instance)
+
+        tags, tags_to_add = self._get_tags(custom_tags, instance)
+
+        if not instance.get(max_slow_entries):
+            max_slow_entries = int(conn.config_get(max_slow_entries)[max_slow_entries])
+            if max_slow_entries > DEFAULT_MAX_SLOW_ENTRIES:
+                self.warning("Redis {0} is higher than {1}. Defaulting to {1}."\
+                    "If you need a higher value, please set {0} in your check config"\
+                    .format(max_slow_entries, DEFAULT_MAX_SLOW_ENTRIES))
+                max_slow_entries = DEFAULT_MAX_SLOW_ENTRIES
+        else:
+            max_slow_entries = int(instance.get(max_slow_entries))
+
+
+        # Generate a unique id for this instance to be persisted across runs
+        ts_key = self._generate_instance_key(instance)
+
+        # Get all slowlog entries
+        
+        slowlogs = conn.slowlog_get(max_slow_entries)
+
+        # Find slowlog entries between last timestamp and now using start_time
+        slowlogs = [s for s in slowlogs if s['start_time'] >
+            self.last_timestamp_seen[ts_key]]
+
+
+        max_ts = 0
+        # Slowlog entry looks like:
+        #  {'command': 'LPOP somekey',
+        #   'duration': 11238,
+        #   'id': 496L,
+        #   'start_time': 1422529869}
+        for slowlog in slowlogs:
+            if slowlog['start_time'] > max_ts:
+                max_ts = slowlog['start_time']
+
+            command_tag = 'command:{0}'.format(slowlog['command'].split()[0])
+            value = slowlog['duration']
+            self.histogram('redis.slowlog.micros', value, tags=tags + [command_tag])
```

Comment:
on second thought, a conversion might be a bad idea because it would make the axis all wonky